### PR TITLE
add various new subproc events

### PR DESCRIPTION
**Pull request by jlipps**
I found that I kept reimplementing these things every time I was using SubProc, so put them into the lib itself

### CODE REVIEW

File: lib/teen_process.js [hunk 1]
```diff
@@ -155,12 +162,9 @@ class SubProcess extends EventEmitter {
...
     // return a promise so we can wrap the async behavior
     return new B((resolve, reject) => {
-      try {
-        // actually spawn the subproc
-        this.proc = spawn(this.cmd, this.args, this.opts);
-      } catch (e) {
-        reject(e);
-      }
+      // actually spawn the subproc
+      this.proc = spawn(this.cmd, this.args, this.opts);
```
Comment (mykola-mokhnach) [inline lib/teen_process.js]: you mean this never throws?

Comment (jlipps) [reply lib/teen_process.js]: if it throws, the promise will be rejected; no need for us to catch and reject specifically

**Comment by jlipps** [comment]
care to have a look @imurchie?

File: lib/teen_process.js (hunk 1 above)
Comment (imurchie) [reply lib/teen_process.js]: I don't even think that it would throw. All error handling in `spawn` is done through the event emitter.

**Comment by jlipps** [comment]
@imurchie FWIW i think i have seen `spawn` throw straightaway, let's say if you try to run a command that doesn't exist.

**Comment by jlipps** [comment]
published in 1.12.0

**Pull request merged**